docs/adr/adr-010-incentivized-testnet-monitoring.md

```diff
@@ -0,0 +1,230 @@
+
+This document proposes a strategy for making data in the Celestia team managed OTEL Collector available for use in internal Grafana dashboards and a public leaderboard. Additionally it describes how a node operator can monitor their own node by deploying their own OTEL Collector instance.
+
+![Incentivized Testnet Monitoring Diagram](./img/incentivized-testnet-monitoring-diagram.svg)
+
+// Comment on diagram <https://lucid.app/lucidchart/d957570f-9c06-4a82-9843-00d8232f734a/edit?invitationId=inv_247247d6-3a67-40b1-8513-1818caadd627>#
+
+### Where to export data to?
+
+Grafana can query data from [multiple data sources](https://grafana.com/docs/grafana/latest/datasources/#supported-data-sources). This document explores two of these data sources:
+
+1. [Prometheus](https://github.com/prometheus/prometheus) is an open-source time series database written in Go. Prometheus uses the [PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) query language. We can deploy Prometheus ourselves or use a hosted Prometheus provider (ex. [Google](https://cloud.google.com/stackdriver/docs/managed-prometheus), [AWS](https://aws.amazon.com/prometheus/), [Grafana](https://grafana.com/go/hosted-prometheus-monitoring/), etc.). Prometheus is pull-based which means services that would like to expose Prometheus metrics must provide an HTTP endpoint (ex. `/metrics`) that a Prometheus instance can poll (see [instrumenting a Go application for Prometheus](https://prometheus.io/docs/guides/go-application/)). Prometheus is used by [Cosmos SDK telemetry](https://docs.cosmos.network/main/core/telemetry.html) and [Tendermint telemetry](https://docs.tendermint.com/v0.35/nodes/metrics.html) so one major benefit to using Prometheus is that metrics emitted by celestia-core, celestia-app, and celestia-node can share the same database.
+2. [InfluxDB](https://github.com/influxdata/influxdb) is another open-source time series database written in Go. It is free to deploy the InfluxDB but there is a commercial offering from [influxdata](https://www.influxdata.com/get-influxdb/) that provides clustering and on-prem deployments. InfluxDB uses the [InfluxQL](https://docs.influxdata.com/influxdb/v1.8/query_language/) query language which appears less capable at advanced queries [ref](https://www.robustperception.io/translating-between-monitoring-languages/). InfluxDB is push-based which means services can push metrics directly to an InfluxDB instance ([ref](https://logz.io/blog/prometheus-influxdb/#:~:text=InfluxDB%20is%20a%20push%2Dbased,and%20Prometheus%20fetches%20them%20periodically.)). See [Prometheus vs. InfluxDB](https://prometheus.io/docs/introduction/comparison/#prometheus-vs-influxdb) for a more detailed comparison.
+
+If alternative data sources should be evaluated, please share them with us.
+
+#### Decision
+
+We agreed to using Prometheus at this time.
```

Comment:
Did we actually do some research that this will work fine with about 1500 light nodes?

Reply:
No but really good question. I'll explore scaling limitations of self-hosted Prometheus, cloud-hosted Prometheus, and OTEL Collector and report back

Reply:
Based on research in [Notion](https://www.notion.so/celestiaorg/Testnet-Telemetry-d85550a3caee4004b00a2e3bf82619b1) it seems possible for us to use Prometheus for the incentivized testnet.

Most articles state that a single Prometheus instance is able to handle millions of time series but I propose we use a cloud-hosted Prometheus instance to avoid the burden of managing and scaling an instance ourselves.

I plan on revising this architecture so that we don't have to maintain a gateway OTEL Collector instance because I couldn't find reliable evidence that it's able to handle receiving data from 1500+ nodes. Stay tuned for an updated revision of this ADR with a new diagram.